recipe/conda_build_config.yaml

```
@@ -11,4 +11,15 @@ channel_priority:
 tiledb:
 - '2.15.*'
 - '2.16.*'
-
+numpy:
+  - 1.20
```

```suggestion
  - 1.21
```

conda-forge is now pinning Python 3.8, 3.9, and 3.10 with numpy 1.21 ([link to conda-forge-pinning](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/53e94822b050849a95e43d87cd916f8b852634c5/recipe/conda_build_config.yaml#L580))

(this will require a rerender afterwards to update the CI yaml files)

---

I'm taking this out entirely and putting only
```
diff --git a/recipe/conda_build_config.yaml b/recipe/conda_build_config.yaml
index 59901fb..8d5be92 100644
--- a/recipe/conda_build_config.yaml
+++ b/recipe/conda_build_config.yaml
@@ -11,4 +11,7 @@ channel_priority:
 tiledb:
 - '2.15.*'
 - '2.16.*'
-
+python:
+  - 3.9.* *_cpython
+  - 3.10.* *_cpython
+  - 3.11.* *_cpython
```

---

(waiting to tag 0.0.9 and include that here to reduce builds)

---

Shouldn't this be handled by `- {{ pin_compatible('numpy') }}`?

---

Below is my recommendation:

```yaml
numpy:
  - 1.21
  - 1.21
  - 1.23
python:
  - 3.9.* *_cpython
  - 3.10.* *_cpython
  - 3.11.* *_cpython
python_impl:
  - cpython
  - cpython
  - cpython
```

Technically you could get away without including `python_impl` in this particular case because you are building for 3 versions of Python, just a different 3 from the upstream conda-forge-pinning. But as this PR demonstrates, we often copy-paste between feedstocks. If anyone were to copy paste it and try to build for 2 or 4 versions of Python, they wouldn't be able to rerender the feedstock